surefire-api/src/main/java/org/apache/maven/surefire/api/report/LegacyPojoStackTraceWriter.java

```diff
@@ -81,10 +79,15 @@ public String smartTrimmedStackTrace()
         result.append( "#" );
         result.append( testMethod );
         SafeThrowable throwable = getThrowable();
+        final String excClassName = throwable.getTarget().getClass().getName();
         if ( throwable.getTarget() instanceof AssertionError )
+        if ( throwable.getTarget() instanceof AssertionError
+                || "junit.framework.AssertionFailedError".equals( excClassName )
+                || "junit.framework.ComparisonFailure".equals( excClassName )
+                || excClassName.startsWith( "org.opentest4j." ) )
         {
             result.append( " " );
-            result.append( getTruncatedMessage( throwable.getMessage(), MAX_LINE_LENGTH - result.length() ) );
+            result.append( throwable.getMessage() );
```

Comment:
See the previous code.
Why you did not see `null` in the console log?
You are missing the null check here!